src/kernels/gpu_reference_kernel/naive_conv.cpp

```diff
@@ -102,6 +103,32 @@ inline __device__ __host__ int8_t cast_to(const int32_t& val)
     return static_cast<int8_t>(val & 0xff);
 }
 
+inline __device__ __host__ bool IsZero(double val) { return fabs(val) < DBL_EPSILON; }
```

Comment:
let's simplify these checks too. We can assume that the user passed exact values for `0.0` and `1.0`.